### PR TITLE
feat: allow images to fill full space of a drawer WEB-181

### PR DESCRIPTION
**Pull request by riffdiktator**
## Jira

[[DS] Update Drawer component to allow images to fill full height of a drawer](https://littlespoon.atlassian.net/browse/WEB-181)

## Motivation

We want to add a feature which allow images to fill up whole height of a drawer

## Current Behavior
Right now whole drawer content is pushed to the bottom. When content is not high enough there's an empty space at the top of a drawer
image

## New Behavior
After adding a new `useFullHeight` prop to the `Drawer` component whole content is stretched to use whole height of a drawer. This feature allow images to fill all empty space at the top

image

## Affected Areas

Drawer component

## Risk Analysis

Low risk

### CODE REVIEW

**Review by huhgawz — APPROVED** [review]

**Comment by huhgawz** [comment]
@riffdiktator Could you paste a screenshot on mobile? 🙏

**Comment by huhgawz** [comment]
@riffdiktator You may want to cherry-pick 9f2783beb295a7b05d991e531f8823124922ff5b

**Comment by riffdiktator** [comment]
> @riffdiktator Could you paste a screenshot on mobile? 🙏

There's no need for that, because this scenario occurs only on wide desktop screens. On mobile drawer is so small that whole space is filled anyway

**Comment by riffdiktator** [comment]
> @riffdiktator You need to cherry-pick [9f2783b](https://github.com/little-spoon-dev/design-system/commit/9f2783beb295a7b05d991e531f8823124922ff5b) to fix tests

It didn't help...

**Comment by huhgawz** [comment]
> > @riffdiktator You need to cherry-pick [9f2783b](https://github.com/little-spoon-dev/design-system/commit/9f2783beb295a7b05d991e531f8823124922ff5b) to fix tests
> 
> It didn't help...

It helped. However, it seems you now need to add some additional tests to increase the coverage: https://github.com/little-spoon-dev/design-system/actions/runs/7395998068/job/20120277427?pr=1098#step:11:1322

File: packages/drawer/__tests__/Drawer.test.tsx [hunk 1]
```diff
@@ -105,6 +105,23 @@ describe('with props.className', () => {
   })
 })
 
+describe('with props.useFullHeight', () => {
+  it('renders a drawer withou useFullHeight prop', () => {
```
Comment (huhgawz) [inline packages/drawer/__tests__/Drawer.test.tsx:109]:
```suggestion
  it('renders a drawer with the default height', () => {
```

File: packages/drawer/__tests__/Drawer.test.tsx [hunk 2]
```diff
@@ -105,6 +105,23 @@ describe('with props.className', () => {
...
+    render(<Drawer open />)
+    const drawerContent = screen.getByTestId('@modal:drawer:content')
+    expect(drawerContent).toBeInTheDocument()
+    const drawerContentStyle = getComputedStyle(drawerContent)
+    expect(drawerContentStyle.flexGrow).toBe('0')
+  })
+  it('renders a drawer with useFullHeight prop', () => {
```
Comment (huhgawz) [inline packages/drawer/__tests__/Drawer.test.tsx:116]:
```suggestion
  it('renders a drawer with full height', () => {
```

**Review by huhgawz — APPROVED** [review]

**Review by akopell — APPROVED** [review]

**Review by huhgawz — APPROVED** [review]
Dev QA done. Added `useFullHeight` property works as expected:

**Pull request merged**